### PR TITLE
Add plumbing for geo point and geo shape fields for the scripting fields api

### PR DESCRIPTION
**Pull request by jdconrad**
This change adds the plumbing for the GeoPoint and GeoShape fields using ToScriptField from the top levels mappings passed through all the way to leaf field data.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/sort/GeoDistanceSortBuilder.java [hunk 1]
```diff
@@ -603,7 +603,11 @@ private IndexGeoPointFieldData fieldData(SearchExecutionContext context) {
         MappedFieldType fieldType = context.getFieldType(fieldName);
         if (fieldType == null) {
             if (ignoreUnmapped) {
-                return new LatLonPointIndexFieldData(fieldName, CoreValuesSourceType.GEOPOINT);
+                return new LatLonPointIndexFieldData(
+                    fieldName,
+                    CoreValuesSourceType.GEOPOINT,
+                    (dv, n) -> { throw new UnsupportedOperationException(); }
```
Comment (stu-elastic) [inline server/src/main/java/org/elasticsearch/search/sort/GeoDistanceSortBuilder.java:609]: Can you add a comment for why this is unsupported?

**Review by stu-elastic — APPROVED** [review]
As expected, just one request.

File: server/src/main/java/org/elasticsearch/search/sort/GeoDistanceSortBuilder.java (hunk 1 above)
Comment (jdconrad) [reply server/src/main/java/org/elasticsearch/search/sort/GeoDistanceSortBuilder.java:610]: Done.

**Comment by jdconrad** [comment]
@stu-elastic Thanks for the review. Will commit as soon as CI passes.

**Pull request merged**